tools/run_tests/xds_k8s_test_driver/bin/cleanup/cleanup.py

```diff
@@ -0,0 +1,253 @@
+    security_td = traffic_director.TrafficDirectorSecureManager(
+        gcp_api_manager,
+        project=project,
+        network=network,
+        resource_prefix=resource_prefix,
+        resource_suffix=resource_suffix)
+    # TODO: cleanup appnet resources.
+    # appnet_td = traffic_director.TrafficDirectorAppNetManager(
+    #     gcp_api_manager,
+    #     project=project,
+    #     network=network,
+    #     resource_prefix=resource_prefix,
+    #     resource_suffix=resource_suffix)
+
+    logger.info('----- Removing traffic director for gke, prefix %s, suffix %s', resource_prefix, resource_suffix)
+    security_td.cleanup(force=True)
+    # appnet_td.cleanup(force=True)
+    plain_td.cleanup(force=True)
+
+
+def main(argv):
+    if len(argv) > 1:
+        raise app.UsageError('Too many command-line arguments.')
+    load_keep_config()
+
+    project: str = xds_flags.PROJECT.value
+    network: str = xds_flags.NETWORK.value
+    dry_run: bool = DRY_RUN.value
+
+    # List resources older than KEEP_PERIOD. We only list health-checks and
+    # instance templates because these are leaves in the resource dependency tree.
+    #
+    # E.g. forwarding-rule depends on the target-proxy. So leaked
+    # forwarding-rule indicates there's a leaked target-proxy (because this
+    # target proxy cannot deleted unless the forwarding rule is deleted). The
+    # leaked target-proxy is guaranteed to be a super set of leaked
+    # forwarding-rule.
+    leakedHealthChecks = exec_gcloud(project, 'compute', 'health-checks', 'list')
+    for resource in leakedHealthChecks:
+        logger.info('-----')
+        logger.info('----- Cleaning up health check %s', resource['name'])
+        if dry_run:
+            # Skip deletion for dry-runs
+            logging.info('----- Skipped [Dry Run]: %s', resource['name'])
+            continue
+
+        # Cleanup resources from the gce framewok.
+        result = re.search(r'test-hc(.*)', resource['name'])
+        if result is not None:
+            if is_marked_as_keep_gce(result.group(1)):
+                logging.info('Skipped [keep]: GCE resource suffix [%s] is marked as keep',
+                             result.group(1))
+                continue
+            remove_relative_resources_run_xds_tests(project, result.group(1))
+            continue
+
+        # Cleanup resources from the gke framework.
+        result = re.search(f'{PSM_SECURITY_PREFIX}-health-check-(.*)',
+                           resource['name'])
+        if result is not None:
+            if is_marked_as_keep_gke(result.group(1)):
+                logging.info('Skipped [keep]: GKE resource suffix [%s] is marked as keep',
+                             result.group(1))
+                continue
+
+            cleanup_td_for_gke(project, network, PSM_SECURITY_PREFIX,
+                               result.group(1))
+            # TODO: cleanup gke clients and servers, but those need k8s context.
```

Comment:
How do I make a k8s context for the clients/servers?
See my attempt commit: https://github.com/menghanl/grpc/commit/eabf3502a2fe8be4e0e90327e48982e7df4b59c0

Reply:
I saw the testing scripts just use `--kube_context="${KUBE_CONTEXT}"`: https://github.com/grpc/grpc-go/blob/master/test/kokoro/psm-security.sh#L103
Is it available for all kokoro jobs?

Reply:
There are 2 pieces of the puzzle. 1) Changing the current `kubectl` context; 2) ask our GKE interop testing framework to use that context.

Reply:
Done